Mathlib/Probability/Moments/SubGaussian.lean

```diff
@@ -725,6 +726,45 @@ end Add
 
 end HasSubgaussianMGF
 
+section HoeffdingLemma
+
+protected lemma mgf_le_of_mem_Icc [IsProbabilityMeasure μ] {a b t : ℝ} (hm : AEMeasurable X μ)
+    (hb : ∀ᵐ ω ∂μ, X ω ∈ Set.Icc a b) (hc : μ[X] = 0) (ht : 0 < t) :
+    mgf X μ t ≤ exp ((‖b - a‖₊ / 2) ^ 2 * t ^ 2 / 2) := by
+  have hi (u : ℝ) : Integrable (fun ω ↦ exp (u * X ω)) μ := integrable_exp_mul_of_mem_Icc hm hb
+  have hs : Set.Icc 0 t ⊆ interior (integrableExpSet X μ) := by simp [hi, integrableExpSet]
+  obtain ⟨u, h1, h2⟩ := exists_iteratedDeriv_two_cgf_mul_eq_cgf ht hc hs
+  rw [← exp_cgf (hi t), exp_le_exp, ← h2]
+  gcongr
+  calc
+  _ = Var[X; μ.tilted (u * X ·)] := by
+    rw [← variance_tilted_mul (hs (Set.mem_Icc_of_Ioo h1))]
+  _ ≤ ((b - a) / 2) ^ 2 := by
+    convert variance_le_sq_of_bounded ((tilted_absolutelyContinuous μ (u * X ·)) hb) _
+    · exact isProbabilityMeasure_tilted (hi u)
+    · exact hm.mono_ac (tilted_absolutelyContinuous μ (u * X ·))
+  _ = (‖b - a‖₊ / 2) ^ 2 := by field_simp
+
+/-- **Hoeffding lemma**. -/
```

Comment:
Please add a sentence that describes what the lemma proves. Also consider adding that lemma to the overview of results in the file docstring.